src/js/containers/account/WithLatestFy.jsx

```diff
@@ -66,6 +73,115 @@ export const useLatestAccountData = () => {
     ];
 };
 
+export const useQueryParams = (params = '') => {
+    const query = new URLSearchParams(useLocation().search);
+    if (!params) {
+        const rtrn = [];
+        [...query.keys()].forEach((key) => {
+            rtrn.push({ [key]: query.get(key) });
+        });
+        return rtrn;
+    }
+    return params.map((param) => ({ [param]: query.get(param) }));
+};
+
+/*
+    * useValidTimeBasedQueryParams
+    * this function enforces validation logic for fiscal years and fiscal periods in the URL as query params
+    * by using the latest submission periods as a reference.
+*/
+export const useValidTimeBasedQueryParams = (currentUrlFy, currentUrlPeriod = null, requiredParams = ['fy', 'period']) => {
```

Comment:
yeah I was kinda thinking the same thing. In Pt III I move the query param helpers to their own file. The reason this guy is here is because it relates to account level data with the submission period stuff.